File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int readIndex = 0;
        int writeIndex = 1;

        int curElement = nums[readIndex];

        while (readIndex < nums.size()) 
        {
            if (nums[readIndex] != curElement) 
            {
                curElement = nums[readIndex];
                nums[writeIndex] = curElement;
                ++writeIndex;
            }

            ++readIndex;
        }

        return writeIndex;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        writeIndex = 1

        curElement = nums[0]

        for readIndex in range(1, len(nums)):
            if nums[readIndex] != curElement:
                curElement = nums[readIndex]
                nums[writeIndex] = curElement
                writeIndex += 1
            
        return writeIndex

        